# * If we specify --pidfile '' a pid file is not created nor used.
# -- Rui Lopes <rui@ruilopes.com>
diff -ru orig/doc/man/twistd-man.html doc/man/twistd-man.html
--- orig/doc/man/twistd-man.html	Sat Dec 27 00:10:02 2003
+++ doc/man/twistd-man.html	Sat Dec 27 00:10:51 2003
@@ -12,7 +12,7 @@
 with the standard process name (the python binary in most cases).
 </dd><dt><strong>-l</strong>, <strong>--logfile</strong><em>&lt;logfile&gt;</em></dt><dd>Log to a specified file, - for stdout (default twistd.log).
 The log file will be rotated on SIGUSR1.
-</dd><dt><strong>--pidfile</strong><em>&lt;pidfile&gt;</em></dt><dd>Save pid in specified file (default twistd.pid)
+</dd><dt><strong>--pidfile</strong><em>&lt;pidfile&gt;</em></dt><dd>Save pid in specified file, '' for not saving the pid (default twistd.pid)
 </dd><dt><strong>--chroot</strong><em>&lt;directory&gt;</em></dt><dd>Chroot to a supplied directory before running (default -- don't chroot).
 Chrooting is done before changing the current directory.
 </dd><dt><strong>-d</strong>, <strong>--rundir</strong><em>&lt;directory&gt;</em></dt><dd>Change to a supplied directory before running (default .)
@@ -33,4 +33,4 @@
 This is free software; see the source for copying conditions.  There is NO
 warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 </p><h2>SEE ALSO<a name="auto6"></a></h2><p>mktap(1)
-</p></div><p><a href="../howto/index.xhtml">Index</a></p><span class="version">Version: 1.1.1</span></body></html>
\ No newline at end of file
+</p></div><p><a href="../howto/index.xhtml">Index</a></p><span class="version">Version: 1.1.1</span></body></html>
diff -ru orig/doc/man/twistd.1 doc/man/twistd.1
--- orig/doc/man/twistd.1	Sat Dec 27 00:10:02 2003
+++ doc/man/twistd.1	Sat Dec 27 00:11:02 2003
@@ -32,7 +32,7 @@
 The log file will be rotated on SIGUSR1.
 .TP
 \fB\--pidfile\fR \fI<pidfile>\fR
-Save pid in specified file (default twistd.pid)
+Save pid in specified file, '' for not saving the pid (default twistd.pid)
 .TP
 \fB\--chroot\fR \fI<directory>\fR
 Chroot to a supplied directory before running (default -- don't chroot).
diff -ru orig/twisted/scripts/twistd.py twisted/scripts/twistd.py
--- orig/twisted/scripts/twistd.py	Sat Dec 27 00:09:55 2003
+++ twisted/scripts/twistd.py	Sat Dec 27 00:17:13 2003
@@ -84,6 +84,8 @@
 """ %  pid)
 
 def removePID(pidfile):
+    if not pidfile:
+        return
     try:
         os.unlink(pidfile)
     except OSError, e:
@@ -153,7 +155,8 @@
     os.chdir(config['rundir'])
     if not config['nodaemon']:
         daemonize()
-    open(config['pidfile'],'wb').write(str(os.getpid()))
+    if config['pidfile']:
+        open(config['pidfile'],'wb').write(str(os.getpid()))
 
 def startApplication(config, application):
     process = service.IProcess(application, None)
